### PR TITLE
Article history must contain complete article version snippets

### PR DESCRIPTION
**Pull request by nlisgo**

### CODE REVIEW

**Comment by thewilkybarkid** [comment]
Looks good. I think we should add a `description` on `versions` to clarify the ordering.

File: src/api.raml [hunk 1]
```diff
@@ -280,7 +280,7 @@ traits:
                 Individual article history.
             get:
                 description: |
-                    Get an article history.
+                    Get an article history, sorted by version number in descending order.
```
Comment (nlisgo) [inline src/api.raml]: @thewilkybarkid, Is this description adequate?

Comment (thewilkybarkid) [reply src/api.raml]: Sorry, I meant in `article-history.v1.yaml` (there isn't a description yet).

File: src/model/article-history.v1.yaml [hunk 2]
```diff
@@ -9,25 +9,12 @@ properties:
         type: string
         format: date
     versions:
+        description: List of articles, sorted by version in descending order
         type: array
```
Comment (nlisgo) [inline src/model/article-history.v1.yaml:13]: @thewilkybarkid How is this description?

Comment (thewilkybarkid) [reply src/model/article-history.v1.yaml:13]: It's a list of 'versions' not 'articles'.

**Comment by thewilkybarkid** [comment]
@lsh-0 are you happy with this change?

File: src/samples/article-history/v1/unpublished-multiple-versions.json [hunk 3]
```diff
@@ -2,8 +2,71 @@
...
+        {
+            "status": "vor",
+            "id": "14107",
+            "version": 3,
+            "type": "research-article",
+            "doi": "10.7554/eLife.14107",
+            "authorLine": "Yongjian Huang et al",
+            "title": "Molecular basis for multimerization in the activation of the epidermal growth factor",
+            "stage": "preview",
+            "published": "2016-03-28T00:00:00Z",
```
Comment (thewilkybarkid) [inline src/samples/article-history/v1/unpublished-multiple-versions.json:14]: ~~Need to lose this.~~

**Comment by lsh-0** [comment]
so - you're replacing what is there with the snippet?

File: src/samples/article-history/v1/unpublished-multiple-versions.json (hunk 3 above)
Comment (thewilkybarkid) [reply src/samples/article-history/v1/unpublished-multiple-versions.json:14]: Sorry, this is fine!

**Comment by nlisgo** [comment]
@lsh-0 that is what I'm hoping for. If we can get this in then journal-cms (and perhaps search) does not need to retain a knowledge of what values should be present in an article snippet. We will just store the data snippet which is available in the /article/{id}/versions response.

**Comment by lsh-0** [comment]
ok. it probably won't make it into lax until end of week

**Comment by thewilkybarkid** [comment]
Happy for this to be merged when Lax is ready.

**Comment by lsh-0** [comment]
it should be ok to merge now because lax and the bot-lax-adaptor use pinned revisions of the api-raml to test against. 

of course - if the journal site is validating incoming article-json against latest version of the raml we will have problems. pinging @giorgiosironi

**Pull request merged**